src/main/haskell/language-kore/test/parser/Data/Kore/Step/StepTest.hs

```diff
@@ -0,0 +1,583 @@
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MonoLocalBinds        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+module Data.Kore.Step.StepTest (stepTests) where
+
+import           Test.Tasty                            (TestTree, testGroup)
+import           Test.Tasty.HUnit                      (testCase)
+
+import           Data.Kore.AST.Common                  (Application (..),
+                                                        AstLocation (..),
+                                                        Id (..),
+                                                        Pattern (ApplicationPattern),
+                                                        SymbolOrAlias (..),
+                                                        Variable)
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.AST.PureToKore              (patternKoreToPure)
+import           Data.Kore.Building.AsAst
+import           Data.Kore.Building.Patterns
+import           Data.Kore.Building.Sorts
+import           Data.Kore.Comparators                 ()
+import           Data.Kore.IndexedModule.MetadataTools (MetadataTools (..))
+import           Data.Kore.MetaML.AST                  (CommonMetaPattern)
+import           Data.Kore.Step.BaseStep
+import           Data.Kore.Step.Step
+import           Data.Kore.Unification.Unifier         (FunctionalProof (..),
+                                                        UnificationProof (..))
+import           Data.Kore.Variables.Fresh.IntCounter
+
+import           Test.Tasty.HUnit.Extensions
+
+v1 :: MetaSort sort => sort -> MetaVariable sort
+v1 = metaVariable "#v1" AstLocationTest
+a1 :: MetaSort sort => sort -> MetaVariable sort
+a1 = metaVariable "#a1" AstLocationTest
+b1 :: MetaSort sort => sort -> MetaVariable sort
+b1 = metaVariable "#b1" AstLocationTest
+x1 :: MetaSort sort => sort -> MetaVariable sort
+x1 = metaVariable "#x1" AstLocationTest
+var_0 :: MetaSort sort => sort -> MetaVariable sort
+var_0 = metaVariable "#var_0" AstLocationTest
+var_1 :: MetaSort sort => sort -> MetaVariable sort
+var_1 = metaVariable "#var_1" AstLocationTest
+top :: MetaSort sort => sort -> CommonMetaPattern
+top sort = asPureMetaPattern $ metaTop sort
+
+stepTests :: TestTree
+stepTests =
+    testGroup
+        "Step Tests"
+        [ singleStepTests
+        , multipleStepPickFirstTests
+        ]
+
+singleStepTests :: TestTree
+singleStepTests =
```

Comment:
could you please, as you did for the the step with axiom tests, also write a brief textual description of each test? i.e., what rules apply, on which term, and what is the result.